Neos.Fusion/Classes/Core/RuntimeConfiguration.php

```diff
@@ -0,0 +1,235 @@
+<?php
+declare(strict_types=1);
+namespace Neos\Fusion\Core;
+
+use Neos\Fusion\Exception;
+use Neos\Flow\Annotations as Flow;
+use Neos\Utility\Arrays;
+
+/**
+ * @Flow\Proxy(false)
+ */
+final class RuntimeConfiguration
+{
+
+    /**
+     * The parsed Fusion configuration
+     *
+     * @var array
+     */
+    private $fusionConfiguration;
+
+    /**
+     * @var array
+     */
+    private $pathCache = [];
+
+    /**
+     * @var \Closure
+     */
+    private $simpleTypeToArrayClosure;
+
+    /**
+     * @var \Closure
+     */
+    private $shouldOverrideFirstClosure;
+
+    public function __construct(array $fusionConfiguration)
+    {
+        $this->fusionConfiguration = $fusionConfiguration;
+
+        $this->simpleTypeToArrayClosure = function ($simpleType) {
+            return $simpleType === null ? null : [
+                '__eelExpression' => null,
+                '__value' => $simpleType,
+                '__objectType' => null
+            ];
+        };
+
+        $this->shouldOverrideFirstClosure = function ($key, $firstValue, $secondValue): bool {
+            return is_array($secondValue) && isset($secondValue['__stopInheritanceChain']);
+        };
+    }
+
+    /**
+     * Get the expanded Fusion configuration for the given path
+     *
+     * @param string $fusionPath
+     * @return array
+     * @throws Exception
+     */
+    public function forPath(string $fusionPath): array
+    {
+        // Fast path if complete Fusion path is in configuration cache
+        if (isset($this->pathCache[$fusionPath])) {
+            return $this->pathCache[$fusionPath]['c'];
+        }
+
+        // Find longest prefix of path that already is in path cache
+        $pathUntilNow = '';
+        $fusionPathLength = strlen($fusionPath);
+        $offset = $fusionPathLength;
+        while (($offset = strrpos($fusionPath, '/', -($fusionPathLength - $offset + 1))) != false) {
```

Comment:
I would naturally rather do an explode here, and peel elements off the end in the loop, that seems easier to understand than all these string operations and is probably less error prone. But fine either way I guess.